scripts/content-utils/card-tree.ts

```diff
@@ -21,7 +21,131 @@ export type CardLeaf = {
     modifiers?: GameWorldModifier | GameWorldModifier[]
 }
 
+/**
+ *
+ * @param tree The CardTree to recursively turn into an array of cards
+ *
+ * NOTE: The following params should only be used internally by the content utils
+ * to connect CardTrees within other structures, like for example CardSequences.
+ *
+ * For actual content, use `tree.conditions` and `tree.modifiers` instead:
+ * @param _startConditions Internal WorldQuery[] to control when to show this CardTree node. Useful when you want to complete the previous node in a CardSequence before starting the CardTree.
+ * @param _endModifiers Internal GameWorldModifier[] to apply to all CardTree leaf nodes. Useful when you want to complete a CardTree within a CardSequence before moving on to the next node in the CardSequence.
+ */
 export function cardsFromTree(
+    tree: Omit<CardTree, "modifiers">,
+    _startConditions?: WorldQuery[],
+    _endModifiers: GameWorldModifier[] = [],
```

Comment:
Instead of marking these params internal by prefixing with `_`, what about if we split `cardsFromTree()` into an internal implementation and an external implementation?

By separating an internal API from the external one, maybe we could hide the internal parameters to simplify for content developers?

```ts

// External API
function cardsFromTree(tree): CardData[] {
    return INTERNAL_cardsFromTree()
}

// Internal API
function INTERNAL_cardsFromTree(tree, _startConditions, _endModifiers): CardData[] {}

```

Reply:
Exactly what I was going to suggest. Though I would prefer not to have the `INTERNAL_`-prefix. By not exporting the function it is clear that it is an internal API.